test_kcidb.py

```diff
@@ -32,6 +33,40 @@ def test_json_output_options():
                     stdout_re="\\{\n  \".*")
 
 
+def test_package_symbols():
+    """
+    Test that package modules are accessible as symbols after importing
+    the package's module (__init__.py).
+
+    Starting with the kcidb package, this test lists all modules under
+    it using pkgutil, imports the package and verifies that each listed
+    module is available as a symbol with type <class 'module'> inside it.
+    It then continues to all sub-packages recursively until everything
+    is verified.
+    """
+    package_path = kcidb.__path__[0]
+    package_name = kcidb.__name__
+    package_modules = [name for _, name, _
+                       in pkgutil.iter_modules([package_path])]
```

Comment:
Note that `pkgutil.iter_modules()` returns a boolean signifying if the module is a package or not, as the third value you're ignoring here. That should help you make the decisions in the loop. That would be easier if you use the `pkgutil.iter_modules()` in the loop directly, instead of storing the module names in a list. Something like this:

```python
for _, name, is_package in pkgutil.iter_modules([package_path]):
    # ...
```

Reply:
Okay